pkg/types/query/helpers.go

```diff
@@ -0,0 +1,328 @@
+package query
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+type ComparatorType string
+
+const (
+	AddressComparator  ComparatorType = "ADDRESS"
+	EventSigComparator ComparatorType = "EVENT"
+	TopicComparator    ComparatorType = "TOPIC"
+)
+
+// SelectIndexedLogs creates a KeyFilter that filters logs for the provided topic values at the specified
+// topic index. Topic value filters are 'OR'ed together.
+func SelectIndexedLogs(
```

Comment:
As an example from my other comment:
```suggestion
func IndexedLogsKeyFilter(
```
Something like this would be more self explanatory, and not mistaken for performing an operation like `Select*` might be.

Reply:
This could work fine. I just don't want to deviate too much from the original name yet. These are intended to be pre-rolled filters that help in the transition from using `LogPoller` directly.

Reply:
Another option would be `query/keyfilter/keyfilter.go`, so `package keyfilter`, with users calling `keyfilter.IndexedLogs(...)` instead of `query.IndexedLogsKeyfilter(...)`

Reply:
I'm not sure if we even want to keep these as long-term names anyway. We've been trying to move away from `Logs` as a term in favor of `Keys`. But for now, naming them as close to the original can be beneficial for ease of transition.

Reply:
Everything exported from this module needs to be thinking long-term, because we will have to stamp v1.0.0 and commit to a stable API.

Reply:
Right now all of these are EMV impl. keywords: logs, events, topics, indexed, words. Its good UX for devs to have naming similar to Log Poller, but as Jordan said we should be thinking long-term here. Not sure how to make best use of these helper functions. In Q3 we will work on improving CR docs and adding some dev guides, one approach that comes to mind is to keep these helper funcs completely chain agnostic and to document how they map to chain specific concepts. Something similar will have to exist anyway and will have to be maintained by BIX later down the line.

Thoughts? @mateusz-sekara @dimkouv